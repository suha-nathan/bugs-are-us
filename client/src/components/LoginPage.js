import React from 'react'
import {Row, Col, Container, Form, Button} from 'react-bootstrap'
import {Link} from "react-router-dom";
import TempHeader from "./TempHeader";

const LoginPage = (props) => {

    return (
        <div className="login-page-container">
            <TempHeader />

            <Container className="d-flex flex-column justify-content-center align-items-center vh-100 ">
                <h1 className="my-0 ">Bugs R Us</h1>
                <Row className="w-100 h-75 login-page-container__content">
                    <Col md={7} className="login-page-container__gif-section"></Col>
                    <Col md={5} className="login-page-container__login-section d-flex justify-content-center flex-column align-items-center">
                        <h2>Log In</h2>
                        <Form.Control name="email" placeholder="Email" className="my-2" />
                        <Form.Control name="password" placeholder="Password" className="my-2" />
                        <Button variant="success" className="my-2">Log In</Button>
                        <p className="login-page-container__login-section__signup-text">Don't have an account? <Link to="/signup">Sign Up Here</Link></p>
                    </Col>
                </Row>
            </Container>
        </div>
    )
}

export default LoginPage