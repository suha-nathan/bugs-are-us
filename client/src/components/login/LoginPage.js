import React, {useState} from 'react'
import {Row, Col, Container, Form, Button} from 'react-bootstrap'
import {Link, Redirect} from "react-router-dom";
import TempHeader from "../shared/TempHeader";
import * as Yup from "yup"
import {useFormik} from "formik"


const LoginPage = ({isAuth, login}) => {
    const [email,setEmail] = useState("")
    const [password,setPassword] = useState("")

    const loginSchema = Yup.object().shape({
        email: Yup.string().email("Invalid Email").required("please enter your email!"),
        password: Yup.string().required("please enter password!")
    })

    const { handleChange, errors, handleSubmit, touched, values } = useFormik({
        initialValues:{
            email: "",
            password: ""
        },
        validationSchema: loginSchema,
        onSubmit: (values,{resetForm}) => {
            // console.log(values)
            login(values)
        }
    })

    if(isAuth){
        // console.log("Redirecting")
        return <Redirect to={"/"}/>
    }

    return (
        <div className="login-page-container">
            <TempHeader />

            <Container className="d-flex flex-column justify-content-center align-items-center vh-100 ">
                <h1 className="my-0 ">Bugs R Us</h1>
                <Row className="w-100 h-75 login-page-container__content">
                    <Col md={7} className="login-page-container__gif-section"></Col>
                    <Col md={5} className="login-page-container__login-section d-flex justify-content-center flex-column align-items-center">

                        <h2>Log In</h2>
                        <form onSubmit={handleSubmit} >
                            <Form.Control
                                name="email"
                                placeholder="Email"
                                value={values.email}
                                className="my-2"
                                onChange={handleChange}
                            />
                            {errors.email && touched.email ?
                                <div>{errors.email}</div>
                                :
                                null
                            }

                            <Form.Control
                                name="password"
                                placeholder="Password"
                                value={values.password}
                                type="password"
                                className="my-2"
                                onChange={handleChange}
                            />
                            {errors.password && touched.password ?
                                <div>{errors.password}</div>
                                :
                                null
                            }

                            <Button
                                variant="primary"
                                className="my-2"
                                type="submit"
                            >Log In</Button>
                        </form>

                        <p className="login-page-container__login-section__signup-text">Don't have an account? <Link to="/signup">Sign Up Here</Link></p>
                    </Col>
                </Row>
            </Container>
        </div>
    )
}

export default LoginPage