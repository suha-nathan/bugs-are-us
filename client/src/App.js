import './App.css';
import React, {useState,useEffect} from "react"
import {BrowserRouter, Redirect, Route, Switch} from "react-router-dom";
import axios from "axios";
import { Alert } from "react-bootstrap"

import DashboardPage from "./components/dashboard/DashboardPage";
import LoginPage from "./components/login/LoginPage";
import SignupPage from "./components/signup/SignupPage";
import CreateBugPage from "./components/create-bug/CreateBugPage";
import BugDetailsPage from "./components/bug-details/BugDetailsPage";
import EditAccountPage from "./components/edit-account/EditAccountPage";
import ProjectsPage from "./components/projects-page/ProjectsPage";
import Header from "./components/shared/Header";
import Layout from "./components/layout/Layout";
import TestDashboardPage from "./components/dashboard/TestDashboardPage";

function App() {
    const [isSignedUp, setSignedUp] = useState(false)
    const [isAuth, setAuth] = useState(false)
    const [user, setUser] = useState({})
    const [errorMessage, setErrorMessage] = useState("")
    const [successMessage, setSuccessMessage] = useState("")

    const [ projectData, setProjectData ] = useState({})

    useEffect(()=>{
        loadUser()
        loadProjectData()
    },[])


    async function loadProjectData() {
        let res = await axios.get('http://localhost:8080/bug/all', {
            headers: {
                'x-auth-token': `Bearer ${localStorage.getItem('token')}`
            }
        })
        // console.log(res)
        setProjectData(res.data)
    }

    async function login(values) {
        try{
            let res = await axios.post("http://localhost:8080/auth/login", values)
            setAuth(true)
            setUser(res.data.user)
            localStorage.setItem("token",res.data.token)
        }catch(e){
            setErrorMessage(e.response.data.message)
            setTimeout(() => {
                setErrorMessage("")
            }, 2000)
        }
    }

    async function signUp(userInfo) {
        try{
            let res = await axios.post("http://localhost:8080/auth/signup", userInfo)
            console.log("signup success")


            // setAuth(true)
            // console.log("signup success")
            // localStorage.setItem("token",res.data.token)
        }catch(e){
            console.log(e.response.data.message)
            setErrorMessage(e.response.data.message)
            setTimeout(() => {
                setErrorMessage("")
            }, 2000)
        }
    }

    async function loadUser() {
        try{
            let res = await axios.get("http://localhost:8080/user",{
                headers:{
                    "x-auth-token" : `Bearer ${localStorage.token}`
                }
            })
            console.log("loading user",res.data.user)
            setUser(res.data.user)
            setAuth(true)
        }catch(e){
            // setErrorMessage(e.response.data.message)
            setAuth(false)
            localStorage.removeItem("token")
        }

    }

    function logOut() {
        setAuth(false)
        localStorage.removeItem("token")
    }

    // console.log(errorMessage)
    // console.log(isAuth)
    // console.log(user)
  return (
    <div className="App">
        <BrowserRouter>
            {errorMessage&& <Alert variant="danger">{errorMessage}</Alert>}
            {successMessage&& <Alert variant="success">{successMessage}</Alert>}

            <Switch>

                <Route path="/login">
                    <LoginPage isAuth={isAuth} login={login}/>
                </Route>

                <Route path="/signup">
                    <SignupPage isSignedUp={isSignedUp} setSignedUp={setSignedUp} signUp={signUp} setSuccessMessage={setSuccessMessage} />
                </Route>

                <Route path="/projects">
                    {isAuth?
                    <Layout user={user} isAuth={isAuth} logOut={logOut}>
                        <ProjectsPage user={user} />
                    </Layout>
                        :
                        <Redirect to="/login"/>
                    }
                </Route>

                <Route path="/bug/create" exact>
                    {isAuth?
                    <Layout user={user} isAuth={isAuth} logOut={logOut}>
                        <CreateBugPage user={user} loadProjectData={loadProjectData} />
                    </Layout>
                        :
                        <Redirect to="/login"/>
                    }
                </Route>

                <Route path="/bug/:id">
                    {isAuth?
                    <Layout user={user} isAuth={isAuth} logOut={logOut}>
                        <BugDetailsPage
                            projectData={projectData}
                            user={user}
                            loadProjectData={loadProjectData}
                        />
                    </Layout>
                    :
                    <Redirect to="/login"/>
                    }
                </Route>

                <Route path="/user/edit">
                    {isAuth?
                    <Layout user={user} isAuth={isAuth} logOut={logOut}>
                        <EditAccountPage user={user} />
                    </Layout>
                        :
                        <Redirect to="/login"/>
                    }

                </Route>

                <Route>
                    {isAuth?
                        <Layout user={user} isAuth={isAuth} logOut={logOut}>
                            <DashboardPage user={user} projectData={projectData} path="/" exact />
                        </Layout>

                        :
                        <Redirect to="/login"/>
                    }
                </Route>
            </Switch>
        </BrowserRouter>

    </div>
  );
}

export default App;
