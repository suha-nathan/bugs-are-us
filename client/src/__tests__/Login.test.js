import Enzyme, { shallow, mount } from 'enzyme'
// import Adapter from 'enzyme-adapter-react-16'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import LoginPage from '../components/login/LoginPage'
import {Button} from "react-bootstrap";
import { render, screen, fireEvent, findByText, waitFor} from '@testing-library/react'
import React from "react";
import '@testing-library/jest-dom/extend-expect'
import { MemoryRouter } from 'react-router-dom'
import App from '../App'
import { Route } from 'react-router-dom'

// It should return an error if no email or password is provided when button is clicked
// It should show an alert if the email or password is wrong when button is clicked
// It should bring you to signup page if "Sign up" is clicked
// It should render upon starting

describe("Login page", () => {

    const isAuth = jest.fn()
    const login = jest.fn()

    let loginButton
    beforeAll(() => {
        render(
            <MemoryRouter>
                <LoginPage />
            </MemoryRouter>
        )

        loginButton = screen.getByRole('button', {
            name: /log in/i
        })
    })

    it("should render upon loading", () => {
        expect(loginButton).toBeInTheDocument()
    })

    it("should show error message if there is no email", async () => {

        render(
            <MemoryRouter>
                <LoginPage />
            </MemoryRouter>
        )

        fireEvent.submit(screen.getByTestId('login-form'))


        await waitFor(async () => {
            expect( screen.getByText( /please enter your email!/)).toBeInTheDocument()
            expect( screen.getByText( /please enter password!/)).toBeInTheDocument()
        } )


    })

    // it("should log you in if email and password is correct", () => {
    //     expect(wrapper.find('Button[type="submit"]').text()).toBe("Log In")
    //     expect(wrapper.find('#login-password').exists()).toBe(true)
    //     expect(wrapper.find('FormControl[name="email"]').exists()).toBe(true)
    //     expect(wrapper.find('FormControl[name="password"]').exists()).toBe(true)
    //
    //
    //
    // })
    //
    // it("should show an alert if email and password is blank", () => {
    //     expect(wrapper.find('Button').text()).toBe('Log In')
    //
    //
    //     console.log(wrapper.find('form').debug())
    //     wrapper.find('Button').simulate('click')
    //     console.log(wrapper.find('form').debug())
    //     // expect(wrapper.find('.emailError')).toHaveLength(1)
    //     // expect(wrapper.find(".emailError").children()).toHaveLength(1)
    //     expect(wrapper.find('.error-message').text()).toBe("please enter email")
    // })
})




