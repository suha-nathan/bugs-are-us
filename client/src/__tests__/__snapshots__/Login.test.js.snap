// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page should render upon loading 1`] = `ShallowWrapper {}`;
